src/iologindata.cpp

```diff
@@ -145,13 +142,13 @@ uint32_t IOLoginData::gameworldAuthentication(const std::string& accountName, co
 	uint32_t accountId = result->getNumber<uint32_t>("id");
 
 	query.str(std::string());
-	query << "SELECT `account_id`, `name`, `deletion` FROM `players` WHERE `name` = " << db.escapeString(characterName);
+	query << "SELECT `account_id`, `name` FROM `players` WHERE `name` = " << db.escapeString(characterName) << " AND `deletion` = 0";
 	result = db.storeQuery(query.str());
 	if (!result) {
 		return 0;
 	}
 
-	if (result->getNumber<uint32_t>("account_id") != accountId || result->getNumber<uint64_t>("deletion") != 0) {
+	if (result->getNumber<uint32_t>("account_id") != accountId) {
```

Comment:
Maybe add
```
<< "AND `account_id` = " << accountId
```
To the query and skip this branch altogether?

Reply:
Thanks, added too :)